### PR TITLE
Added many new table feature reference tables

### PR DESCRIPTION
**Pull request by hntd187**
## Description

Added reference tables for check-constraints, deletion vectors, column mapping, generated columns, cdf, timestamp with no zone and icebergcompatv1

## How was this patch tested?

I ran it locally and inspected the tables for expected output.

## Does this require an update to the documentation?

### CODE REVIEW

File: dat/generated_tables.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
 from datetime import date, datetime, timedelta
 from decimal import Decimal
 from pathlib import Path
-from typing import Callable, List, Tuple
+from typing import Callable, Tuple, List
```
Comment (MrPowers) [inline dat/generated_tables.py:6]: A linter make this change?

File: dat/generated_tables.py [hunk 2]
```diff
@@ -391,15 +393,149 @@ def create_stats_as_struct(case: TestCaseInfo, spark: SparkSession):
...
+    delta_path = str(Path(case.delta_root).absolute())
+    delta_table: DeltaTable = (DeltaTable.create(spark)
+                               .location(delta_path)
+                               .addColumns(df.schema)
+                               .property('delta.enableDeletionVectors', 'true')
+                               .execute())
+
+    df.repartition(1).write.format('delta').mode('append').save(case.delta_root)
+
+    delta_table.delete(col("letter") == "a")
```
Comment (MrPowers) [inline dat/generated_tables.py:426]: Maybe append some more rows here so the tests are a little more reliable?

File: dat/generated_tables.py [hunk 3]
```diff
@@ -391,15 +393,149 @@ def create_stats_as_struct(case: TestCaseInfo, spark: SparkSession):
...
+    df = get_sample_data(spark)
+    delta_path = str(Path(case.delta_root).absolute())
+    (DeltaTable.create(spark)
+     .location(delta_path)
+     .addColumns(df.schema)
+     .property('delta.enableDeletionVectors', 'true')
+     .execute())
+
+    df.repartition(1).write.format('delta').mode('append').save(case.delta_root)
+    spark.sql(f"ALTER TABLE delta.`{delta_path}` ADD CONSTRAINT const1 CHECK (int > 0);")
```
Comment (MrPowers) [inline dat/generated_tables.py:443]: Oh my, this syntax.

**Review by MrPowers — APPROVED** [review]
LGTM.

Might be good to run Black formatting.

File: dat/generated_tables.py (hunk 3 above)
Comment (hntd187) [reply dat/generated_tables.py:443]: this is the only way to add a constraint :-) delta-rs though has it part of the builder methods :-D

**Pull request merged**